crates/ruff_python_parser/src/parser/mod.rs

```diff
@@ -1050,9 +1050,9 @@ impl RecoveryContextKind {
             RecoveryContextKind::Except => p.at(TokenKind::Except),
             RecoveryContextKind::AssignmentTargets => p.at(TokenKind::Equal),
             RecoveryContextKind::TypeParams => p.at_type_param(),
-            RecoveryContextKind::ImportNames => p.at(TokenKind::Name),
+            RecoveryContextKind::ImportNames => p.at_name_or_soft_keyword(),
             RecoveryContextKind::ImportFromAsNames(_) => {
-                matches!(p.current_token_kind(), TokenKind::Star | TokenKind::Name)
+                p.at(TokenKind::Star) || p.at_name_or_soft_keyword()
```

Comment:
> I think I would just call `p.at_name_or_soft_keyword` `p.at_name` to align it with `parse_name`. The downside is that this can be confusing because there's also a `Name` token.

Yeah, I thought of doing that but I found the explicit name better just to avoid this confusion.

> Could we rename `parse_name` to `parse_identifier`?

They both exists because sometimes the parser needs to construct an `ExprName` node while other times it only needs the `Identifier`.